src/legacy/ui/public/agg_types/buckets/date_histogram.js

```diff
@@ -129,7 +129,7 @@ export const dateHistogramBucketAgg = new BucketAggType({
       },
       default: 'auto',
       options: intervalOptions,
-      editor: intervalTemplate,
+      editorComponent: TimeIntervalParamEditor,
```

Comment:
What do you think if we wait for #35134 where I proposed a new type `select`, which already would have `options` with appropriate keys for `EuiSelect` component?
In such case you wouldn't need additional mappings from `{ val: '...', display: '...'}` to `{ value: '...', text: '...'}` format